libgnucash/engine/gnc-date.h

```diff
@@ -221,6 +221,13 @@ gchar* gnc_ctime (const time64 *secs);
  */
 time64 gnc_time (time64 *tbuf);
 
+/** \brief Normalize a strftime/strptime format string
+ *  \param d_fmt: A strftime/strptime format string
+ *  \return A format string without modifiers. This string is allocated
+ * and must be freed by the caller
+ */
+char *normalize_d_fmt (const char *d_fmt);
```

Comment:
If we really wanted to add this to the published API in would have to be gnc_date_normalize_d_fmt(const char *d_fmt). Fortunately it can stay a static because you'll need to copy it in GncDateTimeImpl::format(), see below.